static/app/utils/integrationUtil.tsx

```diff
@@ -228,3 +229,14 @@ export const getAlertText = (integrations?: Integration[]): string | undefined =
         'Update to the latest version of our Slack app to get access to personal and team notifications.'
       );
 };
+
+export const getExternalActorEndpointDetails = (
+  baseEndpoint: string,
+  mapping?: ExternalActorMapping
+): {apiMethod: 'POST' | 'PUT'; apiEndpoint: string} => {
+  const isValidMapping = mapping && mapping?.id;
+  return {
+    apiMethod: isValidMapping ? 'PUT' : 'POST',
+    apiEndpoint: isValidMapping ? `${baseEndpoint}${mapping.id}/` : baseEndpoint,
```

Comment:
Joining URLs like this is a little fragile. For example, we don't guarantee that `baseEndpoint` is passed in with a trailing slash. It's ugly but I'd do:
```tsx
const apiEndpoint = new URL(mapping.id, baseEndpoint).href;
```
https://developer.mozilla.org/en-US/docs/Web/API/URL/URL

Reply:
I'll refactor this, It was what we were doing prior so I kept it but that is much cleaner

Reply:
@mgaeta Just realized this will not work since the endpoints are declared without the hostname attached. The baseEndpoints of `/organizations/.../external-users/` are not compatable with the URL api since they don't have the protocol specified. This is what our API client requires though, so this might be the best we can do at the moment.

Reply:
Then I'd just leave a TODO or something.